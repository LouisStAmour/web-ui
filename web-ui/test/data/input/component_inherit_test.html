<!doctype html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
</head>
<body>
  <element name="x-base" extends="div" constructor="Base">
    <template>
      <div>Here are the lifecycle methods that have been called:</div>
      <pre>{{status}}</pre>
      <div>You should see Base created, Derived created,
           Base inserted, Derived inserted</div>
    </template>
    <script type="application/dart">
      import 'package:web_ui/web_ui.dart';
      class Base extends WebComponent {
        String status = '';
        created() {
          status = '${status}\nBase.created';
        }
        inserted() {
          status = '${status}\nBase.inserted';
        }
      }
    </script>
  </element>
  <element name="x-derived" extends="x-base" constructor="Derived">
    <template><shadow></shadow></template>
    <script type="application/dart">
      import 'package:web_ui/web_ui.dart';

      class Derived extends Base {
        created() {
          super.created();
          status = '${status}\nDerived.created';
        }
        inserted() {
          super.inserted();
          status = '${status}\nDerived.inserted';
        }
      }
    </script>
  </element>
  <div is="x-derived"></div>
  <script type="application/dart">
    import 'common.dart';
    import 'dart:html';
    main() {
      window.postMessage('done', '*');
    }
  </script>
</body>
</html>
