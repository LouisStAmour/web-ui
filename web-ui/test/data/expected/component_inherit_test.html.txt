Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x174
  RenderBlock {HTML} at (0,0) size 800x174
    RenderBody {BODY} at (8,8) size 784x158
      RenderBlock {DIV} at (0,0) size 784x158
        RenderBlock {DIV} at (0,0) size 784x18
          RenderText {#text} at (0,0) size 530x18
            text run at (0,0) width 530: "Here are the lifecycle methods that have been called:"
        RenderBlock {PRE} at (0,34) size 784x90
          RenderText {#text} at (0,0) size 160x90
            text run at (0,0) width 0: " "
            text run at (0,18) width 120: "Base.created"
            text run at (120,18) width 0: " "
            text run at (0,36) width 150: "Derived.created"
            text run at (150,36) width 0: " "
            text run at (0,54) width 130: "Base.inserted"
            text run at (130,54) width 0: " "
            text run at (0,72) width 160: "Derived.inserted"
        RenderBlock {DIV} at (0,140) size 784x18
          RenderText {#text} at (0,0) size 770x18
            text run at (0,0) width 460: "You should see Base created, Derived created, "
            text run at (460,0) width 310: "Base inserted, Derived inserted"
